### PR TITLE
Bug fix in scope processing

### PR DESCRIPTION
**Pull request by libreliu**
Hi,

Thank you for your python library! I've found the library very useful as I'm working on a VCD comparator written in Python.

I've found the library not behaving as expected when I used things like `$dumpvars(0, x, y);` in iVerilog simulator. This commit now adds a "root" scope on top of the internal hierarchy, and fixes the problem in this vcd piece:

$scope module testbench $end
$var reg 3 ! x [2:0] $end
$upscope $end
$scope module testbench $end
$var wire 8 " y [7:0] $end
$upscope $end
$enddefinitions $end

Also, this commit fixed nesting scope like this:
$scope module aaa $end
$var .. $end
$scope module bbb $end
$var .. $end
$upscope $end
$upscope $end
$enddefinitions $end

However, my commits broke backward compatibility a little bit, so if you've found it useful you might need to bump the package version for indication.

Thanks,
Libre Liu

### CODE REVIEW

**Comment by libreliu** [comment]
Sorry, I've forget to update those testcases...
If you've found those fixes useful, I can fix the test cases.

**Comment by Nic30** [comment]
@libreliu I am very glad for your commits, I will look in to this tonight.

I am preparing a json writer which can also log updates in arrays, so it will be possible to display content of memories.
I am planing to use https://github.com/Nic30/d3-wave to visualize a results of simulations in jupyter based notebooks in https://github.com/Nic30/jupyter_widget_hwt (wip, will be finished at the end of month)

File: pyDigitalWaveTools/vcd/parser.py [hunk 1]
```diff
@@ -74,7 +86,17 @@ def __init__(self):
...
+        # ------
+        # $scope module testbench $end
+        # $var reg 3 ! x [2:0] $end
+        # $upscope $end
+        # $scope module testbench $end
+        # $var wire 8 " y [7:0] $end
+        # $upscope $end
+        # $enddefinitions $end
+        # ------
+        self.scope = VcdVarScope("root", None)
```
Comment (Nic30) [inline pyDigitalWaveTools/vcd/parser.py:99]: self.scope is always initialized to this object and the if below does not wok as expected, should there be a `= None` ?
I am missing something?

Comment (libreliu) [reply pyDigitalWaveTools/vcd/parser.py:99]: Sorry, I've been thinking about deleting all the `self.scope is None` processing when I add these lines, but I eventually forgot..
My plan is to delete all the `None` detections in `vcd_upscope` and `vcd_scope` to make the behaviour unified, and also for cases when multiple top level modules appear in the same vcd (which may not supposed to happen, though)

**Pull request merged**